core/unix/signal_linux.c

```diff
@@ -363,15 +363,24 @@ handle_pre_extended_syscall_sigmasks(dcontext_t *dcontext, kernel_sigset_t *sigm
      */
     if (sizemask != sizeof(kernel_sigset_t))
         return false;
+    /* XXX i#3344: we do not support any kind of nesting of p* variants of system calls,
+     * e.g. by executing another one in a signal handler, should the kernel support this.
```

Comment:
I was unsure whether this will work. We copy the modified pre-syscall mask to the app's frame. App executes another p* syscall. Frame has previous mask, process starts over again. So I think you're right and this should work (if even the kernel supports it natively). Removing comment, I will close i#3344 once merged.

Reply:
> I will close i#3344 once merged

Just put "Fixes #3344" here.  At least put "Issue: #3344".

Reply:
Ok